api/types.proto

```diff
@@ -713,3 +727,31 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretValueID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_value_id = 2 [(gogoproto.customname) = "SecretValueID"];
```

Comment:
Ok, so this is a "valid" condition but represents an error. Such tasks should probably fail.

Presumably, we don't allow the deletion of secrets that have an active reference. This will protect against the malicious case.

The analog for the revision would be to not delete an revisions that have an active reference either.